### ISSUE TITLE
Modal doesn't show properly in MB core software

### ISSUE THREAD

**Issue by bphillips201** [mindbody/mbkit#93]
Modal in MB Core software shows the gray overlay but no modal. 

In Consumer Web (user info is hidden with gray boxes):
![image](https://user-images.githubusercontent.com/1147075/79617590-b2c9d480-80bc-11ea-9c33-f1851882b1ca.png)

**Comment by czaas** [comment]
As discussed in the slack thread: this is a z-index override in Core which requires the veil to have 1004 or more value (at least in the pages I loaded it on). 

As we discussed to keep MBKit lean as possible we will not include any application specific CSS overrides as they will vary wildly from page to page. 

--- 

For situations where these components are consumed in existing applications that contain strong CSS opinions I suggest we create a higher order component (HOC) that applies a class name and passes through all props. 

We could create a second library (CoreMBKit) with mbkit components listed as peer dependencies (so the consuming application can choose which version it wants/needs). They all the HOC does is pass in a CSS class name* and add overrides as needed. 

*We need to be sure to pass along any given class name to the HOC and pass along to the component along with spread all props. We could import the type definitions directly from the component in question so the MBCoreKit component would get all the appropriate types for the component. 

```
yarn add @mbkit/modal @mbcorekit/modal
```

```
import {Modal} from '@mbcorekit/modal'

...

<Modal show={bool} className={string}>
</Modal>
```
